packages/validator-ajv6/src/validator.ts

```diff
@@ -158,6 +159,24 @@ export default class AJV6Validator<T = any, S extends StrictRJSFSchema = RJSFSch
     return validationDataMerge<T>({ errors, errorSchema }, userErrorSchema);
   }
 
+  /**
+   * This function checks if a schema needs to be added and if the root schemas don't match it removes the old root schema from the ajv instance and adds the new one.
+   * @param rootSchema - The root schema used to provide $ref resolutions
+   */
+  handleSchemaUpdate(rootSchema: RJSFSchema): void {
+    const rootSchemaId = ROOT_SCHEMA_PREFIX;
+    // add the rootSchema ROOT_SCHEMA_PREFIX as id.
+    // if schema validator instance doesn't exist, add it.
+    // else 'handleRootSchemaChange' should be called if the root schema changes so we don't have to remove and recompile the schema every run.
+    if (this.ajv.getSchema(ROOT_SCHEMA_PREFIX) === undefined) {
+      // TODO restore the commented out `if` above when the TODO in the `finally` is completed
+      this.ajv.addSchema(rootSchema, ROOT_SCHEMA_PREFIX);
```

Comment:
I'm not sure this comment is necessary
```suggestion
    if (this.ajv.getSchema(ROOT_SCHEMA_PREFIX) === undefined) {
      this.ajv.addSchema(rootSchema, ROOT_SCHEMA_PREFIX);
```